data/bdtopo/raw.py

```diff
@@ -10,34 +10,33 @@
 
 def configure(context):
     context.config("data_path")
-    context.config("bdtopo_path", "bdtopo/ADRESSE.shp")
+    context.config("bdtopo_path", "bdtopo/BATIMENT.shp")
 
     context.stage("data.spatial.codes")
 
 def execute(context):
+    
+    # will be used in further version
     df_codes = context.stage("data.spatial.codes")
-    requested_departements = set(df_codes["departement_id"].unique())
 
     df_bdtopo = []
 
-    with context.progress(label = "Loading BD TOPO address registry ...") as progress:
+    with context.progress(label = "Loading BD TOPO residential buildings registry ...") as progress:
         with fiona.open("%s/%s" % (context.config("data_path"), context.config("bdtopo_path"))) as archive:
             for item in archive:
-                code = item["properties"]["CODE_INSEE"]
-
-                if code[:2] in requested_departements or code[:3] in requested_departements:
+                # we check for every buildings if it has residential use (only or mix use)
+                if item['properties']["NB_LOGTS"] is not None and int(item['properties']["NB_LOGTS"]) > 0:
+                   
                     df_bdtopo.append(dict(
-                        geometry = geo.Point(*item["geometry"]["coordinates"]),
-                        commune_id = item["properties"]["CODE_INSEE"],
-                        raw_number = pd.to_numeric(item["properties"]["NUMERO"]),
-                        raw_street = item["properties"]["NOM_1"]
+                        geometry = geo.Point(*item["geometry"]["coordinates"][0][0][0:2]),
```

Comment:
I assume that these are polygons? Then you could just do
`item["geometry"].centroid`

Reply:
makes sens, done